drivers/azure/azure.go

```diff
@@ -384,16 +384,6 @@ func (d *Driver) PreCreateCheck() (err error) {
 		return fmt.Errorf("Virtual Machine with name %s already exists in resource group %q", d.naming().VM(), d.ResourceGroup)
 	}
 
-	// Check if virtual machine exists. An existing virtual machine cannot be updated.
-	log.Debug("Checking if OS Disk already exists.")
-	exists, err = c.OSDiskExists(ctx, d.ResourceGroup, d.naming().OSDisk())
-	if err != nil {
-		return err
-	}
-	if exists {
-		return fmt.Errorf("OS Disk with name %s already exists in resource group %q", d.naming().VM(), d.ResourceGroup)
-	}
-
```

Comment:
Unnecessary as Azure already logs an error statement on VM create if the OS Disk already exists:

```
2020/05/28 16:31:10 [ERROR] NodeController c-lmsh8/m-4fnw5 [node-controller] failed with : Error creating machine: Error in driver during machine creation: Code="ConflictingUserInput" Message="Disk arvind-test-image-custom1-os-disk already exists in resource group ARVIND-TEST. Only CreateOption.Attach is supported.  Target: '/subscriptions/14a619f7-a887-4635-8647-d8f46f92eaac/resourceGroups/ARVIND-TEST/providers/Microsoft.Compute/disks/arvind-test-image-custom1-os-disk'."
```